### PR TITLE
Add DELETE query support

### LINKED ISSUE #1647: Wire up delete series
This is different than drop series. Drop series can take a set of tags, or an ID.

Delete series can additionally take a time range.

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This commit adds query language support for deleting series with a `DELETE` query.

Fixes #1647

/cc @jwilder 

###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

###### Required only if applicable
- [x] [InfluxQL Spec](https://github.com/influxdata/influxdb/blob/master/influxql/README.md) updated

### CODE REVIEW

**Comment by toni-moreno** [comment]
+1

File: tsdb/engine/tsm1/cache.go [hunk 1]
```diff
@@ -344,6 +344,7 @@ func (c *Cache) DeleteRange(keys []string, min, max int64) {
 		c.store[k].filter(min, max)
 		if c.store[k].count() == 0 {
 			delete(c.store, k)
+			continue
```
Comment (jwilder) [inline tsdb/engine/tsm1/cache.go]: This will skip the `c.size` decrement below.  Should add a `c.size -= uint64(origSize)` before `continue`

**Comment by jwilder** [comment]
Needs a `CHANGELOG` update.

File: tsdb/engine/tsm1/cache.go (hunk 1 above)
Comment (benbjohnson) [inline tsdb/engine/tsm1/cache.go]: Fixed in 2cf25a1.

**Comment by benbjohnson** [comment]
@jwilder I forgot to push the `CHANGELOG` update. Fixed and pushed up.

**Comment by jwilder** [comment]
👍

**Comment by beckettsean** [comment]
OMGYES!

@benbjohnson any initial thoughts on how performant this is? Is it something a user could run during normal operations or would it be advisable only against a quiescent database?

What's the difference between `DELETE SERIES cpu` vs. `DROP SERIES cpu`, neither with `WHERE` conditions. In the first case the series is still in the index and in the second it's not?

**Comment by benbjohnson** [comment]
@beckettsean I'm not sure. @jwilder did the tsm1 implementation. He could probably speak on the performance characteristics.

**Comment by jwilder** [comment]
@beckettsean I'm still testing performance.  Should work like `drop series`.

The difference between `DELETE SERIES` and `DROP SERIES` is that `DELETE SERIES` does not remove the series from the index so `SHOW SERIES` will still list it even if you deleted all data.

This behavior is a little odd after playing with this because, if you restart the server, the series are gone.  @pauldix is there reason why `DELETE SERIES` should not remove the series from the index if the series data is fully removed?

**Comment by nemesisdesign** [comment]
Thanks for adding this.

**Pull request merged**